pkg/render/whisker/component.go

```diff
@@ -0,0 +1,279 @@
+	env := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "INFO"},
+		{Name: "CA_CERT_PATH", Value: "/certs/tls.crt"},
+		{Name: "PORT", Value: "7443"},
+	}
+
+	return corev1.Container{
+		Name:            GoldmaneContainerName,
+		Image:           c.goldmaneImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		Env:             env,
+		SecurityContext: securitycontext.NewNonRootContext(),
+	}
+}
+
+func (c *Component) goldmaneService() *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "goldmane",
+			Namespace: WhiskerNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{Port: 7443}},
+			Selector: map[string]string{
+				"k8s-app": WhiskerDeploymentName,
+			},
+		},
+	}
+}
+
+func (c *Component) deployment() *appsv1.Deployment {
+	tolerations := append(c.cfg.Installation.ControlPlaneTolerations, rmeta.TolerateCriticalAddonsAndControlPlane...)
+	if c.cfg.Installation.KubernetesProvider.IsGKE() {
+		tolerations = append(tolerations, rmeta.TolerateGKEARM64NoSchedule)
+	}
+
+	ctrs := []corev1.Container{c.whiskerContainer(), c.whiskerBackendContainer(), c.goldmaneContainer()}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      WhiskerDeploymentName,
+			Namespace: WhiskerNamespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.ToPtr(int32(1)),
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: WhiskerDeploymentName,
+				},
+				Spec: corev1.PodSpec{
+					NodeSelector:       c.cfg.Installation.ControlPlaneNodeSelector,
+					ServiceAccountName: WhiskerServiceAccountName,
+					Tolerations:        tolerations,
+					ImagePullSecrets:   secret.GetReferenceList(c.cfg.PullSecrets),
+					Containers:         ctrs,
+				},
+			},
+		},
+	}
+}
+
+func (c *Component) clusterRoleBinding() *rbacv1.ClusterRoleBinding {
+	return &rbacv1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "ClusterRoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: WhiskerClusterRoleName,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     WhiskerClusterRoleName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      WhiskerClusterRoleName,
+				Namespace: WhiskerNamespace,
+			},
+		},
+	}
+}
+
+func (c *Component) clusterRole() *rbacv1.ClusterRole {
+	policyRules := []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{""},
+			Resources: []string{"configmaps"},
+			Verbs:     []string{"get", "list", "watch", "create", "update", "patch"},
+		},
+		{
+			APIGroups: []string{""},
+			Resources: []string{"secrets"},
+			Verbs:     []string{"get", "list", "watch"},
+		},
+	}
+
+	return &rbacv1.ClusterRole{
```

Comment:
This should be a namespaced Role instead of a ClusterRole

Reply:
Arg darn copy and paste!